pkg/sql/logictest/testdata/logic_test/select

```diff
@@ -843,3 +843,22 @@ RESET large_full_scan_rows;
 # Regression test for #58104.
 statement ok
 SELECT * FROM pg_catalog.pg_attrdef WHERE (adnum = 1 AND adrelid = 1) OR (adbin = 'foo' AND adrelid = 2)
+
+# Tests with SELECT AS clause.
```

Comment:
I recommend placing the parser tests in the parser/testdata directory, not as a logic test.

Reply:
+1 to the parser test - though this one has some value also since it's worth seeing that the column label `of` is correctly shown.

Reply:
I have kept this logic test as is since its helpful to check for the column label. Additionally, I have added a parser test too.